src/DynamoCoreWpf/Views/Core/NodeView.xaml.cs

```diff
@@ -2360,29 +2363,24 @@ private void DisplayNodeContextMenu(object sender, RoutedEventArgs e)
             ViewModel.DynamoViewModel.ExecuteCommand(
                 new DynCmd.SelectModelCommand(nodeGuid, Keyboard.Modifiers.AsDynamoType()));
 
-            var contextMenu = grid.ContextMenu;
-
-            // Stashing any injected MenuItems from the Node View Customization process.
-            if (contextMenu.Items.Count > 0 && NodeViewCustomizationMenuItems.Count < 1)
-            {
-                StashNodeViewCustomizationMenuItems();
-            }
+            StashNodeViewCustomizationMenuItems();
 
             // Clearing any existing items in the node's ContextMenu.
-            contextMenu.Items.Clear();
+            MainContextMenu.Items.Clear();
             nodeContextMenu.Items.Clear();
             NodeContextMenuBuilder.Build(nodeContextMenu, viewModel, NodeViewCustomizationMenuItems);
 
-            nodeContextMenu.DataContext = viewModel;
-            nodeContextMenu.Closed += MainContextMenu_OnClosed;
-            nodeContextMenu.IsOpen = true;
-
+            MainContextMenu = nodeContextMenu;
+            MainContextMenu.DataContext = viewModel;
+            MainContextMenu.Closed += MainContextMenu_OnClosed;
+            MainContextMenu.IsOpen = true;
             e.Handled = true;
         }
 
         private void MainContextMenu_OnClosed(object sender, RoutedEventArgs e)
         {
-            nodeContextMenu.Closed -= MainContextMenu_OnClosed;
+            MainContextMenu.Closed -= MainContextMenu_OnClosed;
+            MainContextMenu.Items.Clear();
             nodeContextMenu.Items.Clear();
```

Comment:
Since you've assigned `nodeContextMenu` to `MainContextMenu`, are not both of these context menus the same? Why then do they need to be cleared twice?